client/vmmCtrlrP_service.go

```diff
@@ -0,0 +1,259 @@
+
+	cont, err := sm.GetViaURL(dnUrl)
+	list := models.VMMControllerListFromContainer(cont)
+
+	return list, err
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsMcastAddrNs(parentDn, tnFvnsMcastAddrInstPName string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tnFvnsMcastAddrInstPName": "%s"
+			}
+		}
+	}`, "vmmRsMcastAddrNs", dn, tnFvnsMcastAddrInstPName))
+
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationvmmRsMcastAddrNs(parentDn string) error {
+	dn := fmt.Sprintf("%s/rsmcastAddrNs", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsMcastAddrNs")
+}
+
+func (sm *ServiceManager) CreateRelationvmmRsAcc(parentDn, vmmUsrAccPDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s",
+				"tDn": "%s"
+			}
+		}
+	}`, "vmmRsAcc", dn, vmmUsrAccPDn))
+
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationvmmRsAcc(parentDn string) error {
+	dn := fmt.Sprintf("%s/rsacc", parentDn)
+	return sm.DeleteByDn(dn, "vmmRsAcc")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVmmCtrlrP(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvmmCtrlrP-[%s]", parentDn, tDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s"
+			}
+		}
+	}`, "vmmRsVmmCtrlrP", dn))
+
+	jsonPayload, err := container.ParseJSON(containerJSON)
+	if err != nil {
+		return err
+	}
+
+	req, err := sm.client.MakeRestRequest("POST", fmt.Sprintf("%s.json", sm.MOURL), jsonPayload, true)
+	if err != nil {
+		return err
+	}
+
+	cont, _, err := sm.client.Do(req)
+	if err != nil {
+		return err
+	}
+	fmt.Printf("%+v", cont)
+
+	return nil
+}
+
+func (sm *ServiceManager) DeleteRelationvmmRsVmmCtrlrP(parentDn, tDn string) error {
+	dn := fmt.Sprintf("%s/rsvmmCtrlrP-[%s]", parentDn, tDn)
+	return sm.DeleteByDn(dn, "vmmRsVmmCtrlrP")
+}
+func (sm *ServiceManager) CreateRelationvmmRsVxlanNs(parentDn, tnFvnsVxlanInstPName string) error {
+	dn := fmt.Sprintf("%s/rsvxlanNs", parentDn)
+	containerJSON := []byte(fmt.Sprintf(`{
+		"%s": {
+			"attributes": {
+				"dn": "%s","tnFvnsVxlanInstPName": "%s"
+
```

Comment:
We usually add "annotation" parameter in the attributes payload with value as "orchestrator:terraform" if relation class support annotation.